Merged changes made outside of Github

plecos opened the pull request:
Changes made outside of Github but referenced in threads on SoureForge have been merged here.

plecos committed:
Merged changes made outside of Github

Changes made outside of Github but referenced in threads on SoureForge have been merged here.

plecos committed:
fix some syntax errors

Fixed some syntax errors

plecos committed:
Updated copyright and .NET Framework text

plecos committed:
Updated files after build 0.0.0.17

plecos committed:
Updated files after build 0.0.0.17

plecos committed:
Updated files after build 0.0.0.17

plecos committed:
Updated files after build 0.0.0.17

Moved to Visual Studio 2015
Removed invalid setup project for now

plecos committed:
Updated copyright info

dhoer commented on logrotate/app.config:
```diff
@@ -1,3 +1,3 @@
-<?xml version="1.0"?>
-<configuration>
-<startup><supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5"/></startup></configuration>
+<?xml version="1.0"?>
```
It looks like this file has invalid chars introduced into it.

dhoer commented on logrotate/Content/README.txt:
```diff
@@ -1,61 +1,77 @@
+
+Release Notes:
+
+0.0.0.17- 13 Jan 2017 (beta) - merge changes from Github contributors
```
Should this stay in beta?

dhoer reviewed (commented):
Looks good overall.  Just a few minor tweaks needed.

plecos commented:
If you want, I can send you a compiled exe of this pull request for testing. If you find it is working well and addresses the issues, I will then merge the pull request and post to sourceforge. 

Let me know...you will need to send me your email address so I can attach the setup program. 

Regards, 
Ken 


From: "Dennis Hoer" <notifications@github.com> 
To: "plecos/logrotatewin" <logrotatewin@noreply.github.com> 
Cc: "Ken Salter" <ken@thesalters.net>, "Author" <author@noreply.github.com> 
Sent: Saturday, January 14, 2017 1:41:53 PM 
Subject: Re: [plecos/logrotatewin] Merged changes made outside of Github (#13) 



@dhoer commented on this pull request. 

Looks good overall. Just a few minor tweaks needed. 


In logrotate/app.config : 
> @@ -1,3 +1,3 @@
-<?xml version="1.0"?>
-<configuration>
-<startup><supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5"/></startup></configuration>
+<?xml version="1.0"?> 


It looks like this file has invalid chars introduced into it. 


In logrotate/Content/README.txt : 
> +
+https://sourceforge.net/projects/logrotatewin/
+
+Requirements:
+
+.NET Framework v4.5 or better
+
+Installation:
+
+Run the setup.exe to install.  
+
+This installation will copy the executable, README.txt, gnu_license.txt, and a sample .conf file to the folder you specify.
+
+Release Notes:
+
+0.0.0.17- 13 Jan 2017 (beta) - merge changes from Github contributors 


Should this stay in beta? 

— 
You are receiving this because you authored the thread. 
Reply to this email directly, view it on GitHub , or mute the thread .

dhoer commented:
I sent you an email.  Question: have you thought about building, unit and integration testing this with appveyor?

plecos commented:
Got your email and sent a reply.  As to your question, I'm not familiar with appveyor.

dhoer commented:
Thanks.  https://www.appveyor.com/ is a windows continuous integration tool that is free to opensource projects.  You check-in a appveyor.yml that describes the steps to build and test your code: https://www.appveyor.com/docs/build-configuration/.

dhoer commented:
I'm having issues with comments.  Logrotatewin sucks up the next uncommented line as part of the comment and throws Unknown directive error at the end, e.g., 
```
# my comment

c:\test.log {
  ...
}
```
Would result in the following output:

```
...
logrotate: read line # my comment c:\test.log {
logrotate: Skipping comment
...
logrotate: Unknown directive }
``` 

Let me know if you would like this written up as another issue.

plecos commented:
Yes please so I can track it better.

dhoer commented:
Done: Comments in conf causes Unknown directive error #14

dhoer commented:
On windows is SCHTASKS the best way to run logrotate periodically?

plecos commented:
Yeah I think so, at this point at least.

dhoer commented:
I think this can be merge...

dhoer commented:
I have the chef cookbook close to release. Just need to finish up the readme.  Here is an example of using appveyor to lint, unit, and integration test on windows 2012r2: https://ci.appveyor.com/project/dhoer/chef-windows-logrotate

dhoer commented:
The only other issue I ran into is that when I have the schtask running every hour and I do not set the `-s c:/path/to/logrotate.status`, it ignores the frequency and will rotate the logs every hour.  When I add it, then it works as expected.  Let me know if you would like an issue on this as well.  

Thanks for getting this out there for me.  It is much appreciated!

dhoer commented:
One more interesting thing to mention.  If the following vars are global (at the top), the logrotate will not keep a test.log file after it rotates, but if I have them inside e.g. c:\test.log { }, then it does.

```
missingok
compress
delaycompress
copytruncate
notifempty
```

I don't know if this is expected behavior or not.

dhoer commented:
This cookbook is now released: https://supermarket.chef.io/cookbooks/windows_logrotate

plecos commented:
Closing this because it is very old.
